.github/workflows/run-and-test-builds.yml

```diff
@@ -56,9 +58,23 @@ jobs:
       - name: Build example app
         run: yarn build
         working-directory: ./canary/apps/${{ matrix.path }}
-      - name: Install cypress
-        run: for n in {1..3}; do yarn install --no-lockfile && break || sleep 5; done
-        working-directory: ./canary/e2e
+      - name: Restore cypress runner from Cache
+        uses: actions/cache@v2
+        id: restore-cypress-cache
+        with:
+          path: ~/.cache/Cypress
+          key: ${{ runner.os }}-canary-cypress-${{ hashFiles('canary/e2e/yarn.lock') }}
+      - name: Install e2e packages
+        uses: ./.github/actions/install-with-retries
+        with:
+          working-directory: ./canary/e2e
+          skip-cypress-binary: ${{ steps.restore-cypress-cache.outputs.cache-hit }}
+      - name: Cache cypress runner
```

Comment:
Wouldn't the above restore step create a new cache if miss or the old one isn't hit

Reply:
Yes, it's required (and I realize this is not the most readable sequence of steps). Restore cypress cache step will **not** create a new cache because cypress runner hasn't been downloaded yet and is unavailable. So the sequence is like this:

1. Workflow will try to restore cypress cache but fails.
2. Since it failed, `yarn install` will run **with** cypress binary installation.
3. Now that cypress binary is available, github workflow will create a new cache entry.

Reply:
Added [fdbe871](https://github.com/aws-amplify/amplify-ui/pull/2121/commits/fdbe871993ba88076c9fbf39e22ab0dde472ca64) to clarify this